pyro/distributions/stable.py

```diff
@@ -18,10 +18,16 @@ def _unsafe_standard_stable(alpha, beta, V, W, coords):
     # Differentiably transform noise via parameters.
     assert V.shape == W.shape
     inv_alpha = alpha.reciprocal()
-    b = beta * (math.pi / 2 * alpha).tan()
-    v = b.atan() + alpha * V
+    half_pi = math.pi / 2
+    eps = torch.finfo(V.dtype).eps
+    # make V belong to the open interval (-pi/2, pi/2)
+    V = V.clamp(min=2 * eps - half_pi, max=half_pi - 2 * eps)
+    ha = half_pi * alpha
+    b = beta * ha.tan()
+    # +/- `ha` term to keep the precision of alpha * (V + half_pi) when V ~ -half_pi
+    v = b.atan() - ha + alpha * (V + half_pi)
     Z = v.sin() / ((1 + b * b).rsqrt() * V.cos()).pow(inv_alpha) \
-        * ((v - V).cos() / W).pow(inv_alpha - 1)
+        * ((v - V).cos().clamp(min=eps) / W).pow(inv_alpha - 1)
```

Comment:
@fritzo Just to confirm: clamp by `eps` will make the result after `pow(inv_alpha-1)` a bit larger than using tiny (when `inv_alpha - 1 ~ 0.01`, it is 0.8 vs 0.4). Is it fine for you?

Reply:
nvm, we should concern grad issue too; using eps looks safer.

Reply:
That's fine with me. I was worried about the gradients: if the power is small and the value is tiny, then gradients can be very large.

Do you think we should similarly clamp the preceding line?
```py
((1 + b * b).rsqrt() * V.cos()).clamp(min=eps).pow(inv_alpha)
```

Reply:
Oh nevermind, I see the preceding line is already relatively safe due to the eps clamping of `V`.